### PR TITLE
Change login id column type to string

### PR DESCRIPTION
**Pull request by paramsingh**
Changed the type to string as discussed on IRC with default values as uuids. The migration script puts the default values for existing users as their user IDs which should make sure that they do not get logged out.

Also create index and add a constraint and rename to column to login_id from user_login_id.

### CODE REVIEW

File: admin/sql/updates/2019-02-13-change-login-id-type.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
+BEGIN;
+
+ALTER TABLE "user" DROP COLUMN user_login_id;
+ALTER TABLE "user" ADD COLUMN login_id TEXT NOT NULL DEFAULT uuid_generate_v4()::text;
+
+UPDATE "user"
+   SET login_id = id::text;
```
Comment (alastair) [inline admin/sql/updates/2019-02-13-change-login-id-type.sql]: nice 🕶

File: admin/sql/updates/2019-02-13-change-login-id-type.sql [hunk 2]
```diff
@@ -0,0 +1,13 @@
+BEGIN;
+
+ALTER TABLE "user" DROP COLUMN user_login_id;
```
Comment (alastair) [inline admin/sql/updates/2019-02-13-change-login-id-type.sql]: is the idea that we'll run both updates in a row while releasing in prod?

File: listenbrainz/db/user.py [hunk 3]
```diff
@@ -104,7 +104,7 @@ def get_by_user_login_id(login_id):
         result = connection.execute(sqlalchemy.text("""
             SELECT {columns}
               FROM "user"
-             WHERE user_login_id = :user_login_id
+             WHERE login_id = :user_login_id
```
Comment (alastair) [inline listenbrainz/db/user.py]: with this change do we also want to change the name of this method? (get_by_user_login_id -> get_by_login_id)

File: admin/sql/updates/2019-02-13-change-login-id-type.sql [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+ALTER TABLE "user" ADD COLUMN login_id TEXT NOT NULL DEFAULT uuid_generate_v4()::text;
+
+UPDATE "user"
+   SET login_id = id::text;
+
+ALTER TABLE "user" user_login_id_key UNIQUE (login_id);
+CREATE UNIQUE INDEX login_id_ndx_user ON "user" (login_id);
```
Comment (alastair) [inline admin/sql/updates/2019-02-13-change-login-id-type.sql]: I believe that as an implementation detail, in postgres unique constraints are implemented by creating a unique index. With a constraint _and_ index on this field, does it mean that there are two indexes? Do we mind?

File: admin/sql/updates/2019-02-13-change-login-id-type.sql (hunk 2 above)
Comment (paramsingh) [reply admin/sql/updates/2019-02-13-change-login-id-type.sql]: the first one is already run, i'll run this while releasing on beta for testing.

File: listenbrainz/db/user.py (hunk 3 above)
Comment (paramsingh) [reply listenbrainz/db/user.py]: right, good point.

File: admin/sql/updates/2019-02-13-change-login-id-type.sql (hunk 4 above)
Comment (paramsingh) [reply admin/sql/updates/2019-02-13-change-login-id-type.sql]: this is how it is done for other columns, i'd keep it as is and change them all if needed.

File: listenbrainz/db/user.py (hunk 3 above)
Comment (alastair) [reply listenbrainz/db/user.py]: Perfect

File: admin/sql/updates/2019-02-13-change-login-id-type.sql (hunk 4 above)
Comment (alastair) [reply admin/sql/updates/2019-02-13-change-login-id-type.sql]: Fine

**Pull request merged**